c7n/mu.py

```diff
@@ -119,18 +120,27 @@ def create(self):
             arc_prefix = os.path.relpath(root, venv_lib_path)
             files = self.filter_files(files)
             for f in files:
-                self._zip_file.write(
-                    os.path.join(root, f),
-                    os.path.join(arc_prefix, f))
+                f_path = os.path.join(root, f)
+                dest_path = os.path.join(arc_prefix, f)
+                self.add_file(f_path, dest_path)
 
     def add_file(self, src, dest):
-        self._zip_file.write(src, dest)
+        info = zipfile.ZipInfo(dest)
+        timestamp = os.path.getmtime(src)
+        info.date_time = datetime.timetuple(
```

Comment:
time.localtime is less churn on temp objects

its worth reviewing the std lib implementation
https://github.com/python/cpython/blob/master/Lib/zipfile.py#L473